src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -1050,5 +1089,23 @@ private static unsafe nint UnalignedCountVector128(ref char searchSpace)
             // isn't too important to pin to maintain the alignment.
             return (nint)(uint)(-(int)Unsafe.AsPointer(ref searchSpace) / ElementsPerByte) & (Vector128<ushort>.Count - 1);
         }
+
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryFindFirstMatchedLane(Vector128<byte> compareResult, ref int matchedLane)
+        {
+            Debug.Assert(AdvSimd.IsSupported);
+
+            ulong matches = AdvSimd.Arm64.MaxPairwise(compareResult, compareResult).AsUInt64().ToScalar();
```

Comment:
I think we can avoid this `MaxPairwise` call by reusing the `AddPairwise` call below and comparing its result to zero.

Reply:
Yes, I didn't want to do `AddPairwise` frequently before checking if there are matches, but turns out that doing a check with `AddPairwise` is better.

Reply:
For reference - 
![image](https://user-images.githubusercontent.com/12488060/84452316-555c8a80-ac0a-11ea-911d-e1b55cfee4d0.png)